### PR TITLE
Rename libp2p_helper in go.nix

### PR DESCRIPTION
**Pull request by georgeee**
Move libp2p helper from `bin/libp2p_helper` to `bin/mina-libp2p_helper` in generated derivation.
This allows importing derivation as `buildInput` without use of environment variable (not used in this PR, but will be used in future work).

Affects only nix, changes effectively nothing in operation of it, just updates some internals.

Explain how you tested your changes:
* Built Mina with nix, all good

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

File: nix/go.nix [hunk 1]
```diff
@@ -74,6 +74,10 @@ final: prev: {
       cp -r --reflink=auto ${final.libp2p_ipc_go}/ vendor/libp2p_ipc
       sed -i 's/.*libp2p_ipc.*//' go.mod
     '';
+    # For compatibility with mina's child_processes library
+    postInstall = ''
+      mv $out/bin/libp2p_helper $out/bin/mina-libp2p_helper
```
Comment (rbonichon) [inline nix/go.nix:79]: Yes! That's nice to be able to avoid some issues automatically

**Review by rbonichon — APPROVED** [review]
Super trivial changes but removes a pain point that I've run into. :100:

**Comment by georgeee** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Pull request merged**